docs/functionalabstractions/concretetypes/index.md

```diff
@@ -3,4 +3,111 @@ id: index
+
+### Hash
+
+The `Hash` abstraction builds on the `Equal` abstraction in a different way, describing a type that can be hashed. Its fundamental representation is as a function `A => Int`.
+
+The `Hash` abstraction allows us to define our own way of hashing data types instead of being forced to use the implementation of `hashCode` for existing data types. It also ensures that the definition of the hash is consistent with the definition of equality, another common source of bugs.
+
+### Debug
+
+The `Debug` abstraction describes a type that can be rendered for debugging purposes. Its fundamental representation is as a function `A => Repr`, where `A` is a structured representation of the value that supports rendering in various formats.
+
+The `Debug` abstraction allows us to define how a data type should be rendered, for example rendering an `Array` to display its values rather than just its memory location. The more structured representation also allows rendering in different formats, for example including fully qualified names in the rendering of the data type so the rendering is itself valid Scala code.
+
+## Ways Of Combining Concrete Types
+
+The second set of abstractions describe different ways of combining two values of a given type. These can all be thought of as functions `(A, A) => A`.
+
+Combining is quite fundamental because it is how we build more complex structures from simpler ones.
+
+Generally unary operators are not very interesting because they don't add additional structure, they just modify existing structure.
+
+In contrast, given the ability to combine two values of a given type we can create a new value of that type with richer structure. And if we can do that once we can do it again and again after that, creating values with arbitrarily complex structure.
+
+The abstractions in ZIO Prelude for describing combining values are each based on fundamental algebraic properties that the combining operation can have.
+
+### Associative
+
+The most basic such abstraction is `Associative`, which describes a combining operation that is associative.
+
+```scala
+(a <> b) <> c === a <> (b <> c)
+```
+
+This essentially means that if we are combining three values then the order of operations doesn't matter. We can combine `a` and `b` and then combine the result with `c` or we can combine `b` and `c` and then combine `a` with the result.
+
+This is quite an important property for being able to build more complex structures from simpler ones. Otherwise the result of sequentially building more complex structures from simpler ones is not well defined because it depends on the order of operations and there are an exponentially increasing number of ways we could order these operations as we combine more and more values.
```

Comment:
If you feel like it, you could mention zipping/tupling suffers from not being associative.

Reply:
It is associative subject to an equivalence relationship and that will be done automatically in ZIO 2.0.